x-pack/legacy/plugins/apm/public/components/shared/TransactionActionMenu/sections.ts

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { AppMountContext } from 'kibana/public';
+import { Location } from 'history';
+import url from 'url';
+import { pick } from 'lodash';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { getInfraHref } from '../Links/InfraLink';
+import { getDiscoverQuery } from '../Links/DiscoverLinks/DiscoverTransactionLink';
+import { getDiscoveryHref } from '../Links/DiscoverLinks/DiscoverLink';
+import { fromQuery } from '../Links/url_helpers';
+import { IUrlParams } from '../../../context/UrlParamsContext/types';
+
+function getInfraMetricsQuery(transaction: Transaction) {
+  const plus5 = new Date(transaction['@timestamp']);
+  const minus5 = new Date(plus5.getTime());
+
+  plus5.setMinutes(plus5.getMinutes() + 5);
+  minus5.setMinutes(minus5.getMinutes() - 5);
+
+  return {
+    from: minus5.getTime(),
+    to: plus5.getTime()
+  };
+}
+
+interface Action {
+  label: string;
+  href: string;
+  condition: boolean;
+}
+
+interface SectionItem {
+  title?: string;
+  subtitle?: string;
+  actions: Action[];
+}
+
+interface Section {
```

Comment:
`SectionRecord` might be more appropriate here. A `Section` would be `SectionItem[]`.